[8.0] web_action_conditionable

ovnicraft opened the pull request:
This module implements condition for actions in tree view.
do allows: edit=“false” create=“false” delete=“false”
This module allows too: delete=“state==‘draft’”, simple expression.

ovnicraft committed:
Web addon

This module implements condition for actions in tree view.
do allows: edit=“false” create=“false” delete=“false”
This module allows too: delete=“state==‘draft’”, simple expression.

pedrobaeza commented:
Add a README according OCA's template (https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst) explaining the usage of the module.

dreispt commented on web_action_conditionable/__openerp__.py:
```diff
@@ -0,0 +1,14 @@
+    'data': ['views/view.xml'],
+    "author": "Cristian Salamea,Odoo Community Association (OCA)",
+    "installable": True,
+    "active": False,
```
The `active`key is deprecated.

dreispt commented:
Needs the README with the module description.

ovnicraft committed:
Add Readme

removed active key.

ovnicraft commented:
@pedrobaeza @dreispt done.

dreispt commented on web_action_conditionable/README.rst:
```diff
@@ -0,0 +1,77 @@
+
+This module was written to extend the functionality of actions in tree views.
+Odoo by default support delete="false" create="false", with this module you can:
+delete="state=='draft'".
```
Use backticks around the code samples.

dreispt commented on web_action_conditionable/README.rst:
```diff
@@ -0,0 +1,77 @@
+Configuration
+=============
+
+No needed.
```
s/No/Not

dreispt commented:
LGTM (no test) :+1:

ovnicraft committed:
Better description

ovnicraft committed:
FIX

testing in all cases is better catch errno in JSON.parse.
attrs[action] is always “string” type.

ovnicraft committed:
Fix flake fail

ovnicraft commented:
@dreispt ok looks all is green :+1:

hbrunn commented on web_action_conditionable/static/src/js/views.js:
```diff
@@ -0,0 +1,23 @@
+      var attrs = this.fields_view.arch.attrs;
+      if (action in attrs) {
+	try {
+	  data = JSON.parse(attrs[action]);
```
don't do that yourself, call super here

ovnicraft replied:
why not ? call super gives any better point here ?

hbrunn replied:
you don't know what happens in the future, and also not what other modules exist that might also do whatever with this function. By calling super and only acting on failures, you give other code the chance to do whatever that code is good for

hbrunn commented on web_action_conditionable/static/src/js/views.js:
```diff
@@ -0,0 +1,23 @@
+	try {
+	  data = JSON.parse(attrs[action]);
+	  return data;
+	} catch(error) {
```
does it make sense to inspect error and only do this if we had a json parse error?

ovnicraft replied:
@hbrunn attrs comes with "true" when developer puts explicit and this module wants to support simple expression to decide.
parse  `state=='draft'` gives syntax error

hbrunn replied:
great, then you can do your parsing when super raises a syntax error

ovnicraft replied:
will try your suggestion

ovnicraft committed:
Use super to allow better future hacking

ovnicraft commented:
@hbrunn sorry for delay, its fixed.

hbrunn commented:
@ovnicraft no worries :+1:
